src/QSymbolicsBase/objects.jl

```diff
@@ -0,0 +1,71 @@
+"""This file defines quantum objects (kets, bras, and operators) with various properties"""
+struct SKet <: Symbolic{AbstractKet}
+    name::Symbol
+    basis::Basis
+end
+struct SBra <: Symbolic{AbstractBra}
+    name::Symbol
+    basis::Basis
+end
+struct SOperator <: Symbolic{AbstractOperator}
+    name::Symbol
+    basis::Basis
+end
+const SymQ = Union{SKet, SBra, SOperator}
+isexpr(::SymQ) = false
+metadata(::SymQ) = nothing
+basis(x::SymQ) = x.basis
+
+symbollabel(x::SymQ) = x.name
+Base.show(io::IO, x::SKet) = print(io, "|$(symbollabel(x))⟩")
+Base.show(io::IO, x::SBra) = print(io, "⟨$(symbollabel(x))|")
+Base.show(io::IO, x::SOperator) = print(io, "$(symbollabel(x))")
+Base.show(io::IO, x::SymQObj) = print(io, symbollabel(x)) # fallback that probably is not great
+
+ishermitian(x::SOperator) = false
+isunitary(x::SOperator) = false
+iscommutative(x::SOperator) = false
+
+"""Inverse Operator"""
+@withmetadata struct SInverseOperator <: Symbolic{AbstractOperator}
+    op::Symbolic{AbstractOperator}
+end
+isexpr(::SInverseOperator) = true
+iscall(::SInverseOperator) = true
+arguments(x::SInverseOperator) = [x.op]
+operation(x::SInverseOperator) = inverse
+head(x::SInverseOperator) = :inverse 
+children(x::SInverseOperator) = [:inverse, x.op]
+basis(x::SInverseOperator) = basis(x.op)
+Base.show(io::IO, x::SInverseOperator) = print(io, "$(x.op)⁻¹")
+inverse(x::Symbolic{AbstractOperator}) = SInverseOperator(x)
+
+"""Hermitian Operator"""
```

Comment:
same question about what this is and about expanding docstring

Reply:
The idea with Hermitian and unitary operators was that one could make a symbolic operator hermitian or unitary with the `hermitian` and `unitary` functions, respectively. Maybe there's a more straightforward way to do this, or it would be clear if I just took the time to expand the docstrings.

Reply:
oh, I see now. Specifically, you have made them `isexpr(::SHermitianOperator) = false`. That makes a lot of sense.

I think I would still prefer if they do not wrap around `SOperator`. For starters, even if we keep the structure you have, `op` should probably be only literal, not arbitrary, i.e. `op::SOperator`.

But I would prefer if there is no wrapping at all, i.e. `SHermitianOperator` and `SOperator` should be basically the same, except for `ishermitian` being defined for the former.

`SInvOperator` should probably be together with `SDagger` though, not here, as it is not a literal.

Reply:
OK, I agree with all of this, will make the updates.

Was also wondering if we wanted to define an operator that was both hermitian and unitary. Should we simply define another structure, say `SHermitianUnitaryOperator`?

Reply:
the "proper" way to do this would be to either:

- Use "traits" or a type parameter or something in the metadata, but that would require a bit more design work
- Set up an "assumptions" system... which is one of the most difficult things for any computer algebra system. The design discussions on topics like this span megabytes of email threads on any dev mailing list (e.g. sympy's)

I would prefer if you spend your time on the lower-hanging fruits, so I would suggest indeed going for `SHermitianUnitaryOperator`. That can always be refactored in a non-breaking manner to use one of the aforementioned "better" approaches.